internal/pkg/generate/generate_scripts.go

```diff
@@ -335,16 +390,32 @@ func EnvFile(gd *GenData, at io.Writer) error {
 			return err
 		}
 	}
+
 	executionApiUrl := gd.ExecutionApiUrl
 	executionAuthUrl := gd.ExecutionAuthUrl
 	if cls[execution] != nil {
+		// modify the EC_API_URL endpoint to use for starknet
 		if executionApiUrl == "" {
-			executionApiUrl = fmt.Sprintf("%s:%v", cls[execution].Endpoint, gd.Ports["ELApi"])
+			var executionEndpoint string
+			if gd.Full || cls[starknet] != nil {
+				endpoint := endpointOrEmpty(cls[execution])
+				if strings.HasPrefix(endpoint, "http") {
+					executionEndpoint = strings.TrimPrefix(endpoint, "http")
+				}
+				gd.ExecutionApiUrl = "ws" + executionEndpoint + ":" + strconv.Itoa(int(gd.Ports["ELApi"]))
+				executionApiUrl = fmt.Sprintf("%s", gd.ExecutionApiUrl)
+			}
```

Comment:
Juno only supports WebSocket connections to the Ethereum execution client. So, the a need to ensure that the generated URL is WebSocket compatible when a Starknet full node is created

Reply:
Would be better if we can have a different parameter for the WebSocket url in that case as the Execution API is used by the other services if defined. Think it will work right now as all clients support either the ws:// or http:// versions. But, think it would be cleaner to have both of them.